src/services/leafletWatchHelpers.js

```diff
@@ -20,7 +20,17 @@ angular.module("leaflet-directive")
   to set watching to once and its watch depth.
   */
   var _maybeWatch = function(scope, thingToWatchStr, watchOptions, initCb){
-      return _maybe(scope, '$watch', thingToWatchStr, watchOptions, initCb);
+      var localWatchOptions, watchMethod;
```

Comment:
I don't want break backwards compat unnecessarily. One option would be to keep 'doWatch' as an explicit boolean for watching a marker or a marker collection. If it's set to false, whatever `type` is set to is ignored. If set to true and `type` is not specified, it defaults to `normal`. If set to true and `type` is specified (`collection`, `normal`, or `deep`), then that is used. `isDeep` could be supported for a while as well (forces type to `deep`).

The directiveControls approach is a good one (potentially better) and I probably would have used it if were around when I started using this directive. The watch options I propose just give an alternative approach that are more in line with the original design the of the directive.  As long as both methods (directiveControls and watchers) of marker/path/geojson updating are supported, I think having the collection option is important. As I said before, I think we could also expose more types of updating options through such an interface (instead of a watch, maybe one an event).